### PR TITLE
Issue #78 add Correlation ID functionality to AsyncAPI

### PR DESCRIPTION
**Pull request by SensibleWood**
As follows:

* Added Correlation ID object
* Added runtime expression support
* Added Correlation ID Map to Components Object
* Added Correlation ID to Message Object
* Updated schema and streetlights example to test

**Would welcome suggestions as to whether more references are needed anywhere else in the specification as it stands i.e. defining a Correlation ID at a higher level than Message**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -121,4 +121,4 @@ Thanks goes to these wonderful people ([emoji key](https://allcontributors.org/d
...
 <!-- ALL-CONTRIBUTORS-LIST:END -->
 
-This project follows the [all-contributors](https://github.com/all-contributors/all-contributors) specification. Contributions of any kind welcome!
\ No newline at end of file
+This project follows the [all-contributors](https://github.com/all-contributors/all-contributors) specification. Contributions of any kind welcome!
```
Comment (fmvilas) [inline README.md]: what?

File: versions/1.2.0/asyncapi.md [hunk 2]
```diff
@@ -1966,10 +2090,84 @@ user_pass: []
...
 
+##### <a name="runtimeExpressionExamples"></a>Examples
 
+Source Location | Example expression  | Notes
+---|:---|:---|
+Message Header Property | `$message.header.correlationId` | Headers are analogous to message properties
+Message Payload Property | `$message.payload.meta.messageId` | Correlation ID is set using a property from the `meta` section of a message
 
+Runtime expressions preserve the type of the referenced value.
+Expressions can be embedded into string values by surrounding the expression with `{}` curly braces.
```
Comment (fmvilas) [inline versions/1.2.0/asyncapi.md]: Let's keep PRs simple and with a single scope, please.

File: versions/1.2.0/schema.json [hunk 3]
```diff
@@ -767,6 +784,35 @@
...
+        },
+        "type": {
+          "type": "string",
+          "description": "An optional type specification",
+          "enum": ["string", "integer", "number"]
+        },
+        "location": {
+          "type": "string",
+          "description": "A runtime expression that specifies the location of the correlation ID",
+          "pattern": "^\\$message\\.(header|payload)\\..*"
```
Comment (fmvilas) [inline versions/1.2.0/schema.json]: ditto

File: versions/1.2.0/schema.json [hunk 4]
```diff
@@ -767,6 +784,35 @@
...
+        }
+      }
+    },
```
Comment (fmvilas) [inline versions/1.2.0/schema.json]: Please, move all the changes in this file to version `next`.

File: versions/1.2.0/asyncapi.md [hunk 5]
```diff
@@ -735,6 +735,7 @@ Field Name | Type | Description
 ---|:---:|---
 <a name="messageObjectHeaders"></a>headers | [Schema Object](#schemaObject) | Definition of the message headers. It MAY or MAY NOT define the protocol headers.
 <a name="messageObjectPayload"></a>payload | [Schema Object](#schemaObject) | Definition of the message payload.
+<a name="messageObjectCorrelationId"></a>correlationId | [Correlation ID Object](#correlationIdObject) \| [Reference Object](#referenceObject) | Definition of the correlation ID used for message tracing or matching
```
Comment (fmvilas) [inline versions/1.2.0/asyncapi.md]: Please, apply these changes to version `next` instead of this one.

File: package-lock.json [hunk 6]
```diff
@@ -744,7 +744,7 @@
     },
     "wrap-ansi": {
       "version": "2.1.0",
-      "resolved": "https://registry.npmjs.org/wrap-ansi/-/wrap-ansi-2.1.0.tgz",
+      "resolved": "http://registry.npmjs.org/wrap-ansi/-/wrap-ansi-2.1.0.tgz",
```
Comment (fmvilas) [inline package-lock.json]: Can you change this back?

**Comment by fmvilas** [comment]
Also, the Correlation ID link is missing in the Index at the beginning of the spec.

**Comment by SensibleWood** [comment]
Going to close and resubmit @fmvilas - looks like my merge was a dud....

**Pull request closed without merging**